xcdat/averager.py

```diff
@@ -20,275 +23,311 @@
 # Using xarrays data reduction functionality gives similar results, but is slower than the averager function defined here
 # Averaging axes are specified with a list (e.g., axis=['lat', 'lon']) rather than the CDAT convention (axis='xy')
 # It would be helpful to be able to get axis weights (get_axis_weights()) more generically (some information is hardcoded)
-    # If .bounds had a method to return a list of bounds entries, we could create weights for all dimensions
+# If .bounds had a method to return a list of bounds entries, we could create weights for all dimensions
 # May want to keep the axis weights or full weight array (i.e., add this information into the variable dataarray properties)
 # Is xvariable a good name?
 # Need to write unit tests
 # Can remove logging around bounds found in dataset
 # Functions are too specific to "lat" and "lon" (may have issues if dims are latitude / longitude)
 # Should remove these notes and outline...
 
+#: List of floats representing user-specified region axis bounds
+RegionAxisBounds = List[float]
+#: Dictionary of axis keys mapped to their bounds
+AxisWeights = Dict[Hashable, xr.DataArray]
+#: List of supported axis for spatial averaging
+SupportedAxes = Literal["lat", "lon"]
+SUPPORTED_AXES: Tuple[SupportedAxes, ...] = get_args(SupportedAxes)
 
-def _validate_region_lon_bounds(grid_lon_bounds, region_lon_bounds):
```

Comment:
I broke this function up into the following:

1. `_validate_region_lon_bounds()`
2. `_swap_lon_region_axis()` (originally `_switch_lon_bounds()`
2. `_is_region_lon_in_domain()`